include/dlaf/matrix/tile.h

```diff
@@ -282,26 +316,27 @@ class Tile : public Tile<const T, device> {
 
   /// Returns the base pointer.
   T* ptr() const noexcept {
-    return memory_view_();
+    return data_.ptr();
   }
 
   /// Returns the pointer to the (i, j)-th element,
   /// where @p i := @p index.row and @p j := @p index.col.
   ///
   /// @pre index.isIn(size()).
   T* ptr(const TileElementIndex& index) const noexcept {
-    return memory_view_(linearIndex(index));
+    return data_.ptr(index);
   }
 
   /// Sets the promise to which this Tile will be moved on destruction.
   ///
   /// @c setPromise can be called only once per object.
   /// @pre this Tile should not be a subtile.
-  Tile& setPromise(promise_t<TileType>&& p) {
-    DLAF_ASSERT(!p_, "setPromise has been already used on this object!");
-    DLAF_ASSERT_HEAVY(!sf_.valid(), "setPromise cannot be used on subtiles!");
-    DLAF_ASSERT_HEAVY(!sfc_.valid(), "setPromise cannot be used on subtiles!");
-    p_ = std::make_unique<promise_t<TileType>>(std::move(p));
+  Tile& setPromise(TilePromise&& p) {
```

Comment:
Would this be an option?

```suggestion
  Tile&& setPromise(TilePromise&& p) {
```

At least some of the uses wrap `setPromise` in `move`, but then some of the tests just ignore the return value.

Reply:
Or even return type `Tile` and then `return std::move(*this)`. Hmm, not sure. I'm basing this on "can be called only once per object" which would make it fitting that the tile is invalidated after calling this. But I may be misunderstanding...

Reply:
Setting a dependency doesn't and shouldn't invalidate the object.

Reply:
Thanks @rasolca. So then if I understand correctly returning `Tile&` from `setPromise` is only to support `move(tile.setPromise(...))`. I'd argue that an explicit
```
tile.setPromise(...);
... move(tile) ...;
```
would be significantly clearer. I found it very surprising that it returns a reference to itself. Is this understanding correct or am I missing some other use of it?

The only place I've seen that kind of an API is if you want to support something like `thing.setA(...).setB(...).setC(...)` but that doesn't seem to be the case here.

Reply:
To add, I realize this is not a change from this PR, so I don't want to block this PR based on this discussion.

Reply:
I agree it is not the best choice, and how I wrote in the note, the API should be re-analyzed and refactored.
However I think it is not worth to do the refactoring now, but to wait for senders/receivers.

Reply:
Yep, agreed, and no worries. Let's come back to this separately.